### PR TITLE
fix(api-reference): improve enum support

### PR DESCRIPTION
**Pull request by soGit**
**Explanation**
Scalar has multiple issues & improvements related to Enum support in query parameters:
https://github.com/scalar/scalar/pull/1212
https://github.com/scalar/scalar/issues/1846
https://github.com/scalar/scalar/issues/2006

Here is a sandbox to check the the current support state. https://sandbox.scalar.com/e/8lwnj

**Solution**
With this PR following improvements was implemented:
- display enum value with a single value enum
- do not display 'enum' if it is an array of enum values
- display up to 5 enum values in a line (display flex) + >5 - (display grid)
- add 'allowed values' for array values if it is enum

Links:
https://github.com/scalar/scalar/discussions/2798

### CODE REVIEW

File: packages/api-reference/src/components/Content/Schema/SchemaProperty.vue [hunk 1]
```diff
@@ -283,6 +300,14 @@ const rules = ['oneOf', 'anyOf', 'allOf', 'not']
   border-radius: var(--scalar-radius-lg);
 }
 
+.property-enum-values {
+  display: flex;
+  gap: 6px;
+}
+.property-enum-values-long {
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
```
Comment (soGit) [inline packages/api-reference/src/components/Content/Schema/SchemaProperty.vue:309]: we might use https://developer.mozilla.org/en-US/docs/Web/CSS/repeat

or define is a better approach, for example. Display enum values as chips or code blocks

**Comment by hanspagel** [comment]
What a great PR! Just a quick note: The first issue you mentioned is fixed already, the second seems to be not relevant.

I’m tagging @cameronrohani here, so he can check how it’s rendered and if it’s what we’re aiming for.

**Comment by soGit** [comment]
> What a great PR! Just a quick note: The first issue you mentioned is fixed already, the second seems to be not relevant.
> 
> I’m tagging @cameronrohani here, so he can check how it’s rendered and if it’s what we’re aiming for.

Sure. I've linked the first issue, because in this Pull Requests presents same improvements related to it.
Thank you for pointing out on the second link, it should be https://github.com/scalar/scalar/issues/1846 (updated in the initial description as well)

**Comment by cameronrohani** [comment]
Hey @soGit Marc and I will take a look in to this one this week 😄

**Comment by hanspagel** [comment]
Man, we can definitely do better. But there’s so much that changed with enums lately, that I have to close this PR. I know that @cameronrohani already looked at your PR and took some inspiration from it. Thanks for investing the time to submit this PR!

**Pull request closed without merging**